### PR TITLE
implement telemetry protocol, add test

### PR DESCRIPTION
**Pull request by noot**
Thank you for contributing to Mina! Please see `CONTRIBUTING.md` if you haven't
yet. In that doc, there are more details around how to start our CI.

Explain your changes here.
- implement telemetry protocol to allow for peers to query other peers for telemetry data, even if remote peer is at max connections.

Explain how you tested your changes here.
`go test ./... -run TestGetTelemetry -v`

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

Closes #0000
Closes #0000

### CODE REVIEW

File: src/app/libp2p_helper/src/libp2p_helper/main.go [hunk 1]
```diff
@@ -1209,6 +1211,43 @@ func (ap *findPeerMsg) run(app *app) (interface{}, error) {
...
+		app.P2p.Logger.Error("failed to open stream: ", err)
+		return nil, err
+	}
+
+	defer func() {
+		_ = s.Close()
+	}()
+
+	var data string
+	dec := json.NewDecoder(s)
```
Comment (imeckler) [inline src/app/libp2p_helper/src/libp2p_helper/main.go:1241]: Can you add a timeout to this? Maybe 5 seconds or something?

File: src/app/libp2p_helper/src/libp2p_helper/main.go [hunk 2]
```diff
@@ -1209,6 +1211,43 @@ func (ap *findPeerMsg) run(app *app) (interface{}, error) {
...
+}
+
+type getPeerTelemetryDataMsg struct {
+	PeerID string `json:"peer_id"`
+}
+
+func (m *getPeerTelemetryDataMsg) run(app *app) (interface{}, error) {
+	// Open a "get telemetry" stream on m.PeerID,
+	// block until you can read the response, return that.
+	s, err := app.P2p.Host.NewStream(app.Ctx, peer.ID(m.PeerID), codanet.TelemetryProtocolID)
```
Comment (imeckler) [inline src/app/libp2p_helper/src/libp2p_helper/main.go:1230]: Does this implicitly "`Connect`" to the peer?

File: src/app/libp2p_helper/src/libp2p_helper/main.go (hunk 1 above)
Comment (imeckler) [inline src/app/libp2p_helper/src/libp2p_helper/main.go:1241]: We should also change this to read a fixed at most a fixed number of bytes (8192 should be enough...)

File: src/app/libp2p_helper/src/codanet.go [hunk 3]
```diff
@@ -514,6 +517,28 @@ func (h *Helper) handlePxStreams(s network.Stream) {
 	}
 }
 
+func (h *Helper) handleTelemetryStreams(s network.Stream) {
+	defer func() {
+		_ = s.Close()
+	}()
+	bz, err := json.Marshal(h.TelemetryData)
```
Comment (imeckler) [inline src/app/libp2p_helper/src/codanet.go:524]: Does it need to be JSON / would it just be cheaper to somehow write the string directly?

Comment (noot) [reply src/app/libp2p_helper/src/codanet.go:524]: the JSON only adds two bytes, the marshalled string `"testdata"` is this: `[34 116 101 115 116 100 97 116 97 34]` it's just easier to code cause it deals w/ the length for you, but would work either way. lemme know if you want me to change it

File: src/app/libp2p_helper/src/libp2p_helper/main.go (hunk 2 above)
Comment (noot) [reply src/app/libp2p_helper/src/libp2p_helper/main.go:1230]: yeah exactly

Comment (noot) [reply src/app/libp2p_helper/src/libp2p_helper/main.go:1230]: note that it'll error if we don't have the peer's addr infos in the peerstore

File: src/app/libp2p_helper/src/libp2p_helper/main.go (hunk 1 above)
Comment (noot) [reply src/app/libp2p_helper/src/libp2p_helper/main.go:1241]: kk, i can change it to not be JSON and just read into a fixed size buffer then

**Review by psteckler — COMMENTED** [review]
Can you add the information requested in the PR template (code purpose, how tested, etc.)?

**Comment by noot** [comment]
@psteckler  done!

**Pull request merged**